src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -185,26 +218,19 @@ public boolean publish(String message, JSONArray attachments, String color) {
         return result;
     }
 
-    private String getTokenToUse() {
-        if (authTokenCredentialId != null && !authTokenCredentialId.isEmpty()) {
-            StringCredentials credentials = lookupCredentials(authTokenCredentialId);
+    private String getTokenToUse(String authTokenCredentialId, String token) {
+        if (!StringUtils.isEmpty(authTokenCredentialId)) {
+            StringCredentials credentials = CredentialsObtainer.lookupCredentials(authTokenCredentialId, null);
```

Comment:
can this have an overloaded method please so we don't have to pass null around?
```suggestion
            StringCredentials credentials = CredentialsObtainer.lookupCredentials(authTokenCredentialId);
```